In [1]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

In [4]:
roadacc = pd.read_csv(r"C:\Users\cherd\OneDrive\Pictures\gela\sir mesiera\uk_road_accident.csv")

In [5]:
roadacc

Index Accident_Severity Accident Date   Latitude  \
0       200701BS64157           Serious      5/6/2019  51.506187   
1       200701BS65737           Serious      2/7/2019  51.495029   
2       200701BS66127           Serious    26-08-2019  51.517715   
3       200701BS66128           Serious    16-08-2019  51.495478   
4       200701BS66837            Slight      3/9/2019  51.488576   
...               ...               ...           ...        ...   
660674  201091NM01760            Slight    18-02-2022  57.374005   
660675  201091NM01881            Slight    21-02-2022  57.232273   
660676  201091NM01935            Slight    23-02-2022  57.585044   
660677  201091NM01964           Serious    23-02-2022  57.214898   
660678  201091NM02142           Serious    28-02-2022  57.575210   

                   Light_Conditions           District Area  Longitude  \
0             Darkness - lights lit  Kensington and Chelsea  -0.209082   
1                          Daylight  Kensington and Chelsea  -0.173647   
2       Darkness - lighting unknown  Kensington and Chelsea  -0.210215   
3                          Daylight  Kensington and Chelsea  -0.202731   
4             Darkness - lights lit  Kensington and Chelsea  -0.192487   
...                             ...                     ...        ...   
660674                     Daylight                Highland  -3.467828   
660675       Darkness - no lighting                Highland  -3.809281   
660676                     Daylight                Highland  -3.862727   
660677       Darkness - no lighting                Highland  -3.823997   
660678                     Daylight                Highland  -3.895673   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
0                          1                   2                     Dry   
1                          1                   2             Wet or damp   
2                          1                   3                     Dry   
3                          1                   4                     Dry   
4                          1                   2                     Dry   
...                      ...                 ...                     ...   
660674                     2                   1                     Dry   
660675                     1                   1            Frost or ice   
660676                     1                   3            Frost or ice   
660677                     1                   2             Wet or damp   
660678                     1                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
0       Single carriageway               Urban     Fine no high winds   
1       Single carriageway               Urban  Raining no high winds   
2                      NaN               Urban                    NaN   
3       Single carriageway               Urban     Fine no high winds   
4                      NaN               Urban                    NaN   
...                    ...                 ...                    ...   
660674  Single carriageway               Rural     Fine no high winds   
660675  Single carriageway               Rural     Fine no high winds   
660676  Single carriageway               Rural     Fine no high winds   
660677  Single carriageway               Rural     Fine no high winds   
660678    Dual carriageway               Rural  Snowing no high winds   

                                Vehicle_Type  
0                                        Car  
1                                        Car  
2                      Taxi/Private hire car  
3       Bus or coach (17 or more pass seats)  
4                              Other vehicle  
...                                      ...  
660674                                   Car  
660675                                   Car  
660676                                   Car  
660677                 Motorcycle over 500cc  
660

In [7]:
roadacc.describe()

Latitude      Longitude  Number_of_Casualties  Number_of_Vehicles
count  660654.000000  660653.000000         660679.000000       660679.000000
mean       52.553866      -1.431210              1.357040            1.831255
std         1.406922       1.383330              0.824847            0.715269
min        49.914430      -7.516225              1.000000            1.000000
25%        51.490690      -2.332291              1.000000            1.000000
50%        52.315641      -1.411667              1.000000            2.000000
75%        53.453452      -0.232869              1.000000            2.000000
max        60.757544       1.762010             68.000000           32.000000

In [20]:
kac_ac = roadacc[roadacc['District Area'] == "Kensington and Chelsea"].describe()
kac_ac

Latitude    Longitude  Number_of_Casualties  Number_of_Vehicles
count  2799.000000  2799.000000           2799.000000         2799.000000
mean     51.497737    -0.187237              1.136120            1.759914
std       0.011786     0.017784              0.497721            0.595030
min      51.477465    -0.226487              1.000000            1.000000
25%      51.488894    -0.200906              1.000000            1.000000
50%      51.495570    -0.186906              1.000000            2.000000
75%      51.505573    -0.173500              1.000000            2.000000
max      51.529692    -0.149820             11.000000            9.000000

In [17]:
west_ac = roadacc[roadacc['District Area'] == "Westminster"]
west_ac

Index Accident_Severity Accident Date   Latitude  \
51      200701BS70035            Slight    28-01-2019  51.527693   
105     200701BS70092            Slight    24-02-2019  51.501749   
131     200701BS70127            Slight    14-02-2019  51.501910   
171     200701BS70173            Slight    26-03-2019  51.520533   
212     200701BS70218            Slight    29-04-2019  51.527785   
...               ...               ...           ...        ...   
532986  201001TB00647            Slight     3/10/2022  51.511055   
533004  201001TB00669            Slight    29-09-2022  51.520511   
533006  201001TB00671            Slight     9/10/2022  51.511714   
533008  201001TB00673            Slight     2/10/2022  51.517462   
533070  201001TB00765            Slight    17-10-2022  51.520945   

             Light_Conditions District Area  Longitude  Number_of_Casualties  \
51      Darkness - lights lit   Westminster  -0.215879                     3   
105     Darkness - lights lit   Westminster  -0.160555                     2   
131                  Daylight   Westminster  -0.159396                     3   
171                  Daylight   Westminster  -0.200447                     1   
212     Darkness - lights lit   Westminster  -0.216020                     1   
...                       ...           ...        ...                   ...   
532986  Darkness - lights lit   Westminster  -0.129195                     1   
533004               Daylight   Westminster  -0.181565                     1   
533006               Daylight   Westminster  -0.188401                     1   
533008               Daylight   Westminster  -0.182119                     1   
533070  Darkness - lights lit   Westminster  -0.180539                     1   

        Number_of_Vehicles Road_Surface_Conditions           Road_Type  \
51                       3                     Dry  Single carriageway   
105                      2                     Dry    Dual carriageway   
131                      2                     Dry  Single carriageway   
171                      1                     Dry  Single carriageway   
212                      2                     Dry  Single carriageway   
...                    ...                     ...                 ...   
532986                   1                     Dry  Single carriageway   
533004                   2             Wet or damp  Single carriageway   
533006                   1                     Dry  Single carriageway   
533008                   2                     Dry  Single carriageway   
533070                   2                     Dry          Roundabout   

       Urban_or_Rural_Area     Weather_Conditions  \
51                   Urban     Fine no high winds   
105                  Urban     Fine no high winds   
131                  Urban     Fine no high winds   
171                  Urban     Fine no high winds   
212                  Urban     Fine no high winds   
...                    ...                    ...   
532986               Urban     Fine no high winds   
533004               Urban  Raining no high winds   
533006               Urban     Fine no high winds   
533008               Urban     Fine no high winds   
533070               Urban     Fine no high winds   

                                Vehicle_Type  
51                     Taxi/Private hire car  
105                                      Car  
131                                      Car  
171     Bus or coach (17 or more pass seats)  
212                                      Car  
...                                      ...  
532986                                   Car  
533004                                   Car  
533006                                   Car  
533008                                   Car  
533070   Van / Goods 3.5 tonnes mgw or under  

[5706 rows x 14 columns]

<h1>Questions</h1>
<br>
<p>1. How many fatal accidents happened in a Dry road condition? </p>
<p>2. What road condintion has the most frequent accidents? </p>
<p>3. What is the most common accident severity? </p>

<h2>1. How many fatal accidents happened in a Dry road condition?</h2>

In [37]:
fatal = roadacc[roadacc['Accident_Severity'] == "Fatal"]['Road_Surface_Conditions'].value_counts()
fatal

Road_Surface_Conditions
Dry                     5788
Wet or damp             2620
Frost or ice             193
Snow                      35
Flood over 3cm. deep      23
Name: count, dtype: int64

<h3>Insights: A total of 5788 fatal accident</h3>


<h2>2. What road condintion has the most frequent accidents?</h2>

In [41]:
frequent_accident = roadacc['Road_Surface_Conditions'].value_counts()
frequent_accident

Road_Surface_Conditions
Dry                     447821
Wet or damp             186708
Frost or ice             18517
Snow                      5890
Flood over 3cm. deep      1017
Name: count, dtype: int64

<h3>Insights: Dry road surface has the most accidents</h3>

<h2>3. What is the most common accident severity?</h2>

In [85]:
roadacc["Accident_Severity"].value_counts()

Accident_Severity
Slight     563801
Serious     88217
Fatal        8661
Name: count, dtype: int64

<h3>Insights: Slight severity is the most common accident</h3>

<h2>4. How many casualties are there in total?</h2>

In [86]:
roadacc["Number_of_Casualties"].sum()

np.int64(896568)

<h3>Insights: There are a total of 896,568 casualties</h3>

<h2>5. What is the average number of casualties per accident?</h2>

In [87]:
roadacc["Number_of_Casualties"].mean()

np.float64(1.357040257068864)

<h3>Insights: There are a total of 1.357040257068864 casualties per accident</h3>

<h2>6. What area has the most accidents?</h2>

In [89]:
roadacc["Urban_or_Rural_Area"].value_counts()

Urban_or_Rural_Area
Urban          421663
Rural          238990
Unallocated        11
Name: count, dtype: int64

<h3>Insights: Urban has the most accidents</h3>

<h2>7. Which district has the most accidents?</h2>

In [90]:
roadacc["District Area"].value_counts()

District Area
Birmingham            13491
Leeds                  8898
Manchester             6720
Bradford               6212
Sheffield              5710
                      ...  
Berwick-upon-Tweed      153
Teesdale                142
Shetland Islands        133
Orkney Islands          117
Clackmannanshire         91
Name: count, Length: 422, dtype: int64

<h3>Insights: Birmingham district has the most accidents</h3>

<h2>8. What is the most common weather condition during accidents?</h2>

In [93]:
roadacc["Weather_Conditions"].value_counts()

Weather_Conditions
Fine no high winds       520885
Raining no high winds     79696
Other                     17150
Raining + high winds       9615
Fine + high winds          8554
Snowing no high winds      6238
Fog or mist                3528
Snowing + high winds        885
Name: count, dtype: int64

<h3>Insights: Fine no high winds is the most common weather condition during accidents</h3>

<h2>9. Which road type has the most accidents?</h2>

In [95]:
roadacc["Road_Type"].value_counts()

Road_Type
Single carriageway    492143
Dual carriageway       99424
Roundabout             43992
One way street         13559
Slip road               7041
Name: count, dtype: int64

<h3>Insights: The road type that has the most accidents is a Single carriageway</h3>

<h2>10. Which vehicle type is most involved in accidents?</h2>

In [97]:
roadacc["Vehicle_Type"].value_counts()

Vehicle_Type
Car                                      497992
Van / Goods 3.5 tonnes mgw or under       34160
Bus or coach (17 or more pass seats)      25878
Motorcycle over 500cc                     25657
Goods 7.5 tonnes mgw and over             17307
Motorcycle 125cc and under                15269
Taxi/Private hire car                     13294
Motorcycle over 125cc and up to 500cc      7656
Motorcycle 50cc and under                  7603
Goods over 3.5t. and under 7.5t            6096
Other vehicle                              5637
Minibus (8 - 16 passenger seats)           1976
Agricultural vehicle                       1947
Pedal cycle                                 197
Data missing or out of range                  6
Ridden horse                                  4
Name: count, dtype: int64

<h3>Insights: A car is the most vehicle involved in accidents</h3>

<h2>11. What is the average number of vehicles involved per accident?</h2>

In [98]:
roadacc["Number_of_Vehicles"].mean()

np.float64(1.8312554205597575)

<h3>Insights: The average number of vehicles involved per accident is 1.8312554205597575</h3>

<h2>12. Which disctrict has the highest total casualties?</h2>

In [110]:
roadacc.groupby("District Area")["Number_of_Casualties"].sum()

District Area
Aberdeen City    1508
Aberdeenshire    2529
Adur              817
Allerdale        1663
Alnwick           351
                 ... 
Wychavon         1943
Wycombe          2442
Wyre             1706
Wyre Forest      1340
York             2453
Name: Number_of_Casualties, Length: 422, dtype: int64

<h3>Insights: Some districts, like Aberdeenshire, have very high total casualties compared to others.</h3>

<h2>13. What is the average number of casualties per accident by district?</h2>

In [112]:
roadacc.groupby("District Area")["Number_of_Casualties"].mean()

District Area
Aberdeen City    1.139834
Aberdeenshire    1.310363
Adur             1.319871
Allerdale        1.474291
Alnwick          1.512931
                   ...   
Wychavon         1.427627
Wycombe          1.412377
Wyre             1.378029
Wyre Forest      1.382869
York             1.293094
Name: Number_of_Casualties, Length: 422, dtype: float64

<h3>Insights: There are some districts that have a higher average per accident</h3>

<h2>14. What is the average number of casualties per accident by weather condition?</h2>

In [114]:
roadacc.groupby("Weather_Conditions")["Number_of_Casualties"].mean()

Weather_Conditions
Fine + high winds        1.386018
Fine no high winds       1.350480
Fog or mist              1.452948
Other                    1.354869
Raining + high winds     1.416641
Raining no high winds    1.408214
Snowing + high winds     1.418079
Snowing no high winds    1.341776
Name: Number_of_Casualties, dtype: float64

<h3>Insights: Fog or mist have a higher average number of casualties than the others</h3>

<h2>15. What is the average number of casualties per accident by light condition?</h2>

In [115]:
roadacc.groupby("Light_Conditions")["Number_of_Casualties"].mean()

Light_Conditions
Darkness - lighting unknown    1.306447
Darkness - lights lit          1.369374
Darkness - lights unlit        1.331105
Darkness - no lighting         1.541657
Daylight                       1.340309
Name: Number_of_Casualties, dtype: float64

<h3>Insights: Darkness or no lighting causes more accidents</h3>

<h2>16. How many fatal accidents by district?</h2>

In [116]:
roadacc[roadacc["Accident_Severity"]=="Fatal"]["District Area"].value_counts()

District Area
Birmingham                  105
Leeds                        93
Highland                     88
East Riding of Yorkshire     85
Bradford                     71
                           ... 
Havant                        2
Orkney Islands                2
Harlow                        1
Clackmannanshire              1
Stevenage                     1
Name: count, Length: 422, dtype: int64

<h3>Insights: Birmingham has the highest number of fatal accidents among all districts.</h3>

<h2>17. Which weather condition has the most fatal accidents?</h2>

In [119]:
roadacc[roadacc["Accident_Severity"]=="Fatal"]["Weather_Conditions"].value_counts()

Weather_Conditions
Fine no high winds       7100
Raining no high winds     848
Fine + high winds         175
Other                     165
Raining + high winds      145
Fog or mist                82
Snowing no high winds      36
Snowing + high winds        3
Name: count, dtype: int64

<h3>Insights: Fine no high winds has the most fatal accidents than the other conditions</h3>

<h2>18. How many fatal accidents by vehicle type?</h2>

In [120]:
roadacc[roadacc["Accident_Severity"]=="Fatal"]["Vehicle_Type"].value_counts()

Vehicle_Type
Car                                      6577
Van / Goods 3.5 tonnes mgw or under       467
Motorcycle over 500cc                     339
Bus or coach (17 or more pass seats)      325
Goods 7.5 tonnes mgw and over             216
Motorcycle 125cc and under                189
Taxi/Private hire car                     155
Motorcycle over 125cc and up to 500cc     105
Motorcycle 50cc and under                  95
Other vehicle                              70
Goods over 3.5t. and under 7.5t            67
Minibus (8 - 16 passenger seats)           29
Agricultural vehicle                       21
Pedal cycle                                 6
Name: count, dtype: int64

<h3>Insights: Cars has the most fatal accidents</h3>

<h2>19. How many slight accidents by weather condition?</h2>

In [121]:
roadacc[roadacc["Accident_Severity"]=="Slight"]["Weather_Conditions"].value_counts()

Weather_Conditions
Fine no high winds       441739
Raining no high winds     69380
Other                     15184
Raining + high winds       8209
Fine + high winds          7134
Snowing no high winds      5637
Fog or mist                2963
Snowing + high winds        773
Name: count, dtype: int64

<h3>Insights: Just like fatal accidents, Fine no high winds has the most slight accidents than the other conditions</h3>

<h2>20. Which vehicle type has the most serious accidents?</h2>

In [122]:
roadacc[roadacc["Accident_Severity"]=="Serious"]["Vehicle_Type"].value_counts()

Vehicle_Type
Car                                      66461
Van / Goods 3.5 tonnes mgw or under       4554
Motorcycle over 500cc                     3457
Bus or coach (17 or more pass seats)      3373
Goods 7.5 tonnes mgw and over             2321
Motorcycle 125cc and under                2031
Taxi/Private hire car                     1771
Motorcycle 50cc and under                 1014
Motorcycle over 125cc and up to 500cc     1014
Goods over 3.5t. and under 7.5t            857
Other vehicle                              767
Agricultural vehicle                       282
Minibus (8 - 16 passenger seats)           276
Pedal cycle                                 39
Name: count, dtype: int64

<h3>Insights: The vehicle type that has the most serious  accidents are cars</h3>